worker/uniter/operation/runhook.go

```diff
@@ -174,15 +180,25 @@ func (rh *runHook) afterHook(state State) (bool, error) {
 		logger.Debugf("unit %v has started but has not yet set status", ctx.UnitName())
 		// We've finished the start hook and the charm has not updated its
 		// own status so we'll set it to unknown.
-		err = rh.runner.Context().SetUnitStatus(jujuc.StatusInfo{
+		err = ctx.SetUnitStatus(jujuc.StatusInfo{
 			Status: string(status.Unknown),
 		})
+	case hooks.RelationBroken:
+		var isLeader bool
+		isLeader, err = ctx.IsLeader()
```

Comment:
check error.
also, if !isLeader, you should just break here. no point getting the relation and so on

Reply:
done